### PR TITLE
Fix: Resubmit PR #1719 from organization account

### PR DESCRIPTION
**Pull request by p-osman**
## Summary by Sourcery

Add a new USTA threat intelligence automation module and connector for retrieving compromised credential events and forwarding them to Sekoia.io intakes.

New Features:
- Introduce a UstaClient SDK wrapper for the USTA threat-stream API with error handling and pagination helpers.
- Add the UstaAtpConnector that periodically polls USTA for compromised credential tickets and forwards them as raw events to the platform.
- Define USTA module and connector configuration models and expose the module entry point for running the connector.
- Provide a manifest describing the USTA integration for Sekoia.io, including configuration schema and metadata.

Enhancements:
- Set up project structure, packaging, and code-style configuration for the new USTA module, including poetry, Dockerfile, and tooling configuration files.

Documentation:
- Introduce an initial changelog documenting the first 1.0.0 release of the USTA connector module.

Tests:
- Add pytest fixtures and connector tests covering happy-path ingestion, empty-result handling, and configuration validation for API key and polling interval.

### CODE REVIEW

File: USTA/usta_modules/usta_atp_connector.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    """
+    Simple Sekoia.io connector that:
+      - Fetches compromised credentials tickets from USTA
+      - Uses `created` as the cursor
+      - Forwards raw JSON events to the intake
+    No extras, retries, metrics, or optionals.
+    """
+
+    module = UstaModule
+    configuration: UstaATPModuleConfiguration
```
Comment (squioc) [inline USTA/usta_modules/usta_atp_connector.py]:
```suggestion
    configuration: UstaATPConnectorConfiguration
```

File: USTA/manifest.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+  "configuration": {
+    "title": "UstaModuleConfig",
+    "description": "Configuration for USTA.",
+    "type": "object",
+    "properties": {
+      "api_key": {
+        "title": "Api Key",
+        "type": "string"
+      }
```
Comment (squioc) [inline USTA/manifest.json]: If the `api_key` follows a pattern, you can enforce its verification in the jsonschema (see [Regular expression](https://json-schema.org/understanding-json-schema/reference/regular_expressions)).
Otherwise, to avoid empty string, you can use the following pattern `^\\S+$` to enforce non-empty string:
```suggestion
      "api_key": {
        "title": "Api Key",
        "type": "string",
        "pattern": "^\\S+$"
      }
```

File: USTA/usta_modules/usta_sdk.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+        # Basic network error handling
+        if response.status_code >= 500:
+            raise UstaAPIError(f"Server error {response.status_code}: {response.text}")
+
+        if response.status_code >= 400:
+            try:
+                detail = response.json()
+            except ValueError:
+                detail = response.text
+            raise UstaAPIError(f"Client error {response.status_code}: {detail}")
```
Comment (squioc) [inline USTA/usta_modules/usta_sdk.py]: Can you handle the HTTP code statuses 401 and 403 with a specific exception?
The purpose would be to catch this specific exception in the connector and log the error as critical. With a critical error, the connector will stop immediately (example: [raise of the exception](https://github.com/SEKOIA-IO/automation-library/blob/main/Sophos/sophos_module/client/auth.py#L102-L104) and [how to handle it](https://github.com/SEKOIA-IO/automation-library/blob/main/Sophos/sophos_module/trigger_sophos_edr_events.py#L100-L102))

**Review by squioc — CHANGES_REQUESTED** [review]
